flang/include/flang/Optimizer/Transforms/Passes.td

```diff
@@ -551,6 +551,17 @@ def SimplifyFIROperations : Pass<"simplify-fir-operations", "mlir::ModuleOp"> {
       "Prefer expanding without using Fortran runtime calls.">];
 }
 
+def ConvertComplexPow : Pass<"convert-complex-pow", "mlir::func::FuncOp"> {
```

Comment:
This must be a pass on the whole module because adding the runtime functions mutates the module. Otherwise MLIR will run the pass concurrently on different functions and there can be race conditions between checking if a function declaration has been added to the module and actually adding it.